validation/td-json-schema-validation.json

```diff
@@ -152,6 +152,10 @@
         }
       ]
     },
+    "bcp47_string":{
+      "type":"string",
+      "pattern": "^(((([A-Za-z]{2,3}(-([A-Za-z]{3}(-[A-Za-z]{3}){0,2}))?)|[A-Za-z]{4}|[A-Za-z]{5,8})(-([A-Za-z]{4}))?(-([A-Za-z]{2}|[0-9]{3}))?(-([A-Za-z0-9]{5,8}|[0-9][A-Za-z0-9]{3}))*(-([0-9A-WY-Za-wy-z](-[A-Za-z0-9]{2,8})+))*(-(x(-[A-Za-z0-9]{1,8})+))?)|(x(-[A-Za-z0-9]{1,8})+)|((en-GB-oed|i-ami|i-bnn|i-default|i-enochian|i-hak|i-klingon|i-lux|i-mingo|i-navajo|i-pwn|i-tao|i-tay|i-tsu|sgn-BE-FR|sgn-BE-NL|sgn-CH-DE)|(art-lojban|cel-gaulish|no-bok|no-nyn|zh-guoyu|zh-hakka|zh-min|zh-min-nan|zh-xiang)))$"
```

Comment:
did you take this definition from an official specification

Reply:
I took it from https://github.com/project-open-data/project-open-data.github.io/blob/master/schema/1_0_final/single_entry.json#L223 and validated it against some picks from https://tools.ietf.org/search/bcp47#appendix-A

I could not find a copy pastable pattern at the IETF spec